www/brud06/sp2/js/play.js

```diff
@@ -0,0 +1,172 @@
+            framerate: 8,
+            repeat: -1,
+        });
+
+
+
+        this.createWorld();
+    }
+
+    update() {
+        //game logic
+        this.physics.collide(this.player, this.walls);
+        this.physics.collide(this.walls, this.enemies);
+        //this.player.angle++;
+        this.movePlayer();
+        if (this.player.y < 0 || this.player.y > 600) {
+            this.playerDie();
+        }
+        if (this.physics.overlap(this.player, this.coin)) {
+            this.takeCoin();
+        }
+        if (this.physics.overlap(this.player, this.enemies)) {
+            this.playerDie();
+        }
+        this.updateEnemySpawn();
+
+    }
+    movePlayer() {
+        if (this.arrow.left.isDown) {
+            this.player.body.velocity.x = -200;
+            this.player.anims.play('left', true);
+        }
+        else if (this.arrow.right.isDown) {
+            this.player.body.velocity.x = 200;
+            this.player.anims.play('right', true);
+        }
+        else {
+            this.player.body.velocity.x = 0;
+            this.player.setFrame(0);
+        }
+        if (this.arrow.up.isDown && this.player.body.onFloor()) {
+            this.player.body.velocity.y = -400;
+            this.jumpSound.play();
+        }
+    };
+    //creates the level
+    createWorld() {
+        this.walls = this.physics.add.staticGroup();
+
+        this.walls.create(10, 170, 'wallV');
+        this.walls.create(10, 340, 'wallV');
+        this.walls.create(10, 500, 'wallV');
+
+        this.walls.create(790, 170, 'wallV');
+        this.walls.create(790, 340, 'wallV');
+        this.walls.create(790, 500, 'wallV');
+
+        this.walls.create(150, 10, 'wallH');
+        this.walls.create(650, 10, 'wallH');
+        this.walls.create(150, 590, 'wallH');
+        this.walls.create(650, 590, 'wallH');
+
+        this.walls.create(300, 150, 'wallH');
+        this.walls.create(500, 150, 'wallH');
+        this.walls.create(300, 450, 'wallH');
+        this.walls.create(500, 450, 'wallH');
+        this.walls.create(50, 300, 'wallH');
+        this.walls.create(750, 300, 'wallH');
+    }
+    playerDie() {
+        this.deadSound.play();
+
+        this.scene.start('menu', { score: this.score });
+    }
+    takeCoin() {
+        this.score += 1;
+        this.coin.destroy();
+        this.scoreLabel.setText('Score: ' + this.score);
+        this.updateCoinPosition();
+        this.coinSound.play();
+
+        this.coin.setScale(0);
+        this.tweens.add({
+            targets: this.coin,
+            scale: 1,
+            duration: 300,
+        });
+        this.tweens.add({
+            targets: this.player,
+            scale: 1.3,
+            duration: 100,
+            yoyo: true,
+        });
+    }
+    //Fixed coin positions to prevent it from spawning in walls
+    updateCoinPosition() {
+        let positions = [
+            { x: 100, y: 60 },

```

Comment:
jenom 6 moznych pozic pro coiny? a proc si to definujete porad dokola a pritom stejne berete jen jednu z nich